cultivation/scripts/flashcore/cli/main.py

```
@@ -0,0 +1,223 @@
+    console.print(f"Processed {len(flashcards)} flashcards. Ingesting into database...")
+
+    try:
+        with FlashcardDatabase() as db:
+            db.initialize_schema()
+            inserted_count = db.upsert_cards_batch(flashcards)
+        console.print(
+            f"[green]Successfully ingested/updated {inserted_count} flashcards into the database.[/green]"
+        )
+    except FlashcardDatabaseError as e:
+        console.print(f"[bold]A database error occurred during ingestion: {e}[/bold]")
+        raise typer.Exit(code=1)
```

Ruff (0.11.9)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)